multiwallet.go

```diff
@@ -429,6 +434,60 @@ func (mw *MultiWallet) LinkExistingWallet(walletName, walletDataDir, originalPub
 	})
 }
 
+// walletsCopy is concurrently-safe way to read from mw.wallets map.
+func (mw *MultiWallet) walletsCopy() map[int]*Wallet {
+	result := make(map[int]*Wallet, len(mw.wallets))
+
+	mw.walletsMu.RLock()
+	for key, value := range mw.wallets {
+		result[key] = value
+	}
+	mw.walletsMu.RUnlock()
+
+	return result
+}
+
+// walletsUpdate is concurrently-safe way to update mw.wallets map.
+func (mw *MultiWallet) walletsUpdate(key int, newWallet *Wallet) {
+	mw.walletsMu.Lock()
+	mw.wallets[key] = newWallet
+	mw.walletsMu.Unlock()
+}
+
+// walletsUpdate is concurrently-safe way to delete from mw.wallets map.
+func (mw *MultiWallet) walletsDelete(key int) {
+	mw.walletsMu.Lock()
+	delete(mw.wallets, key)
+	mw.walletsMu.Unlock()
+}
```

Comment:
adding update/delete methods here instead of inlining lock-unlock mostly for conformance with `walletsReadCopy` func